src/geodesic.rs

```diff
@@ -716,15 +715,14 @@ impl Geodesic {
                 a12 = sig12.to_degrees();
                 omg12 = lam12 / (self._f1 * dnm);
             } else {
-                let mut numit = 0;
                 let mut tripn = false;
                 let mut tripb = false;
                 let mut salp1a = self.tiny_;
                 let mut calp1a = 1.0;
                 let mut salp1b = self.tiny_;
                 let mut calp1b = -1.0;
                 let mut domg12 = 0.0;
-                while numit < self.maxit2_ {
+                for numit in 0..self.maxit2_ {
```

Comment:
This is strictly cosmetic, right? 

(I like both the iteration change and changing `&mut calp` -> `calp`, just making sure I understand)

Reply:
Neither the iteration change nor the switch away from `&mut calp` is purely cosmetic, though I do consider both to also be cosmetic improvements.

For the iteration change, the previous version would increment numit partway through the loop body, such that tests involving numit in the second part of the loop would behave differently.

For the question of whether to pass calp1 by reference, I think the previous version was misinterpreting the python variation. Both the python and C++ counterpart modify the calp1 argument value, but because the argument is passed by value in both cases, it doesn't change the value held by the caller.

Reply:
Indeed - thanks for explaining! 👍 

Notably, the previous transcription was true to the python impl, but agreed it makes sense to move towards the c++ impl here.